tests/Adapters/Microsoft.Bot.Builder.Adapters.Facebook.Tests/FacebookAdapterTests.cs

```diff
@@ -0,0 +1,189 @@
+        {
+            var facebookAdapter = new FacebookAdapter(new Mock<FacebookClientWrapper>(_testOptions).Object);
+
+            Task BotsLogic(ITurnContext turnContext, CancellationToken cancellationToken)
+            {
+                return Task.CompletedTask;
+            }
+
+            await Assert.ThrowsAsync<ArgumentNullException>(async () => { await facebookAdapter.ContinueConversationAsync(null, BotsLogic); });
+        }
+
+        [Fact]
+        public async void ContinueConversationAsyncShouldFailWithNullLogic()
+        {
+            var facebookAdapter = new FacebookAdapter(new Mock<FacebookClientWrapper>(_testOptions).Object);
+            var conversationReference = new ConversationReference();
+
+            await Assert.ThrowsAsync<ArgumentNullException>(async () => { await facebookAdapter.ContinueConversationAsync(conversationReference, null); });
+        }
+
+        [Fact]
+        public async void ContinueConversationAsyncShouldSucceed()
+        {
+            var callbackInvoked = false;
+
+            var facebookAdapter = new FacebookAdapter(new Mock<FacebookClientWrapper>(_testOptions).Object);
+            var conversationReference = new ConversationReference();
+            Task BotsLogic(ITurnContext turnContext, CancellationToken cancellationToken)
+            {
+                callbackInvoked = true;
+                return Task.CompletedTask;
+            }
+
+            await facebookAdapter.ContinueConversationAsync(conversationReference, BotsLogic);
+            Assert.True(callbackInvoked);
+        }
+
+        [Fact]
+        public async System.Threading.Tasks.Task DeleteActivityAsyncShouldThrowNotImplementedException()
+        {
+            var adapter = new FacebookAdapter(new FacebookClientWrapper(_testOptions));
+
+            var activity = new Activity();
+            var turnContext = new TurnContext(adapter, activity);
+
+            var conversationReference = new ConversationReference();
+
+            await Assert.ThrowsAsync<NotImplementedException>(() => adapter.DeleteActivityAsync(turnContext, conversationReference, default));
+        }
+
+        [Fact]
+        public async void ProcessAsyncShouldThrowExceptionWithInvalidBody()
+        {
+            var clientWrapper = new Mock<FacebookClientWrapper>(_testOptions);
+            clientWrapper.Setup(x => x.VerifySignature(It.IsAny<HttpRequest>(), It.IsAny<string>())).Returns(true);
+
+            var facebookAdapter = new FacebookAdapter(clientWrapper.Object);
+
+            var stream = new MemoryStream(Encoding.UTF8.GetBytes("wrong-formatted-json"));
+
+            var httpRequest = new Mock<HttpRequest>();
+            httpRequest.SetupGet(req => req.Query[It.IsAny<string>()]).Returns("test");
+            httpRequest.SetupGet(req => req.Body).Returns(stream);
+            var httpResponse = new Mock<HttpResponse>();
+            var bot = new Mock<IBot>();
```

Comment:
What is the criteria for spacing? Shouldn't all the vars be together? the same question goes for the other tests

Reply:
The idea was (excepting in these lines above where it's mostly careless) to declare a mock variable, and next setup it's methods. 
But it's ok, I'll group all variable declarations together, I think it will be better for legibility